apache/trafficcontrol
# Add csv support to stats_over_http

## What does this PR (Pull Request) do?
This uses the same parsing method as astats csv and provides almost double the performance with less than half the allocations of the json version.

Also add a null check for dsStats. Its possible to do a request to ATS to a DS that does not exist, ATS will then insert stats for that request into the remap stats list. When it gets to TM though it will not be in the list of known DSs so if we try to index by it it will end up null


- [x] This PR fixes #REPLACE_ME OR is not related to any Issue 


## Which Traffic Control components are affected by this PR?


- Traffic Monitor

## What is the best way to verify this PR?


- Add health.polling.format = stats_over_http to a server profile
- Set http_polling_format for TM to be "text/csv"
- Check if stats are pulled correctly in TM

## If this is a bug fix, what versions of Traffic Control are affected?



## The following criteria are ALL met by this PR




## Additional Information

Benchmarks comparing all stats parsing, both astats and stats over http with json and csv. In both CSV instances they are ~100% faster with less than half as many allocations

BenchmarkAstatsJson-8   	    5276	    202209 ns/op	  134453 B/op	    2028 allocs/op
BenchmarkAstatsCSV-8    	    9963	    111798 ns/op	  110001 B/op	     798 allocs/op
BenchmarkStatsJson-8    	    5512	    227104 ns/op	  134991 B/op	    2082 allocs/op
BenchmarkStatsCSV-8     	   10000	    108320 ns/op	  111217 B/op	     837 allocs/op
